homework/collections/src/main/java/com/cbfacademy/CollectionsAssignment.java

```diff
@@ -80,12 +116,34 @@ public static ArrayList<Integer> inBoth(Collection<Integer> ints1, Collection<In
      * @return the most frequently occurring String
      */
     public static String mostFrequent(List<String> list) {
+
+        // No nested loops or non-enhanced for-loops are allowed.
+        String word;
+        String highestOccurence = "";
+        int noOfOccurence;
+        int max = 0;
+        int count = 0;
+        Map<String, Integer> newMap = new HashMap<>();
         // You should solve this problem in two stages: First iterate through
         // the list to count occurrences of each String. Then iterate through
+        for (int i = 0; i < list.size(); i++) {
+            word = list.get(i);
+            noOfOccurence = Collections.frequency(list, word);
+
+            newMap.put(word, noOfOccurence);
+        }
```

Comment:
This a regular (non-enhanced) for loop, which shouldn't be used as per the guide comment. You may want Google how an enhanced for loop is implemented

Reply:
ok thank you. By enhanced for loop, do you mean for..each loop?

Reply:
Generally, it's better to use `git add [filename]` as you can very easily otherwise stage files that you're not intending to commit. You can stage multiple files using `git add [file1] [file2] ... [fileN]`; only use `git add .` when you know for sure that you want to include everything in the folder.